### ISSUE TITLE
Enhancement: [no-unnecessary-condition] Make allowConstantLoopConditions more granular

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#7047]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unnecessary-condition

### Description

The [`allowConstantLoopConditions` option](https://typescript-eslint.io/rules/no-unnecessary-condition/#allowconstantloopconditions) is disabled by default, because cases like `while (shouldRun = true)` are likely errors. But that means the rule option also flags cases like `while (true)`. Which is more of a stylistic preference, I think.

How about making the option more granular:

* `'never'`: same as today's `false`
* `'only-simple'`: allow "simple" cases like `true` _(a better name would be nice!)_
* `'always'`: same as today's `true`

...and in future major version, breaking change to:

* current: `'never'` is be the default value and aliased by `false` _(to not be a breaking change)_
* next: `'only-simple'` becomes the default value

### Fail

```typescript
let shouldRun = true;

while (shouldRun = true) { /* ... */ }
```

### Pass

```typescript
while (true) { /* ... */ }
```

### Additional Info

I stumbled onto this in https://github.com/JoshuaKGoldberg/template-typescript-node-package/pull/469/files#diff-e1ce717e1179a0db14e90ec5374768a206651ca807db58352991eb7895ed7c9cR64

**Comment by danvk** [comment]
I like @JoshuaKGoldberg's suggestion of breaking up this rule in order to change the default. It would definitely be nice to continue flag this error:

```
- while (shouldRun = true) { /* ... */ }
+ while (shouldRun == true) { /* ... */ }
```

That being said, @bradzacher's suggestion of setting `allowConstantLoopConditions` to `true` by default would be easier and wouldn't preclude adopting Josh's idea in a future, non-breaking change. Maybe the term could be `only-literal`? or `only-true`?

The current state of the world is bad because this rule bans `while (true)` by default but [the docs](https://typescript-eslint.io/rules/no-unnecessary-condition/#allowconstantloopconditions) say nothing about why this is considered bad or what the suggested alternative is. So far as I can tell there is none, the only alternative is to set `allowConstantLoopConditions`.

**Comment by Josh-Cena** [comment]
FYI the alternative is `for (;;)` because it's fully syntactic and we are not specifying any constant values.
